### PR TITLE
LPD-35560 Migrate PreviewStyleBookOnPages to playwright

### PR DESCRIPTION
**Pull request by pat270**
https://liferay.atlassian.net/browse/LPD-35560

New pr to replace https://github.com/liferay-platform-experience/liferay-portal/pull/776. Something went wrong with CI.

### CODE REVIEW

**Comment by pat270** [comment]
ci:test:sf

**Comment by pat270** [comment]
ci:test:playwright

**Comment by pat270** [comment]
ci:test:sf

**Comment by pat270** [comment]
ci:test:playwright

**Comment by pat270** [comment]
ci:test:sf

**Comment by pat270** [comment]
@thiago-buarqque The playwright test failures are unrelated. 

1. FAIL on Master - Can switch template with web content on widget page.
2. FAIL on Master - Assert the items in the actions ellipsis are displayed
3. PASS on Master/This PR - Content Flags, Content Ratings and Content Display are compatible with Collection Display
4. FAIL on Master - Can select a navigation menu from existing ones
5. FAIL on Master - Multishipping tab displays correctly when enabled
6. PASS on Master/This PR - Verify email notification checkbox is displayed
7. FAIL on Master - Order Details - Questions & Answers
8. FAIL on Master - As a order manager with buyer approval workflow, I can approve orders on pending orders page

File: modules/test/playwright/pages/style-book-web/StyleBooksPage.ts [hunk 1]
```diff
@@ -84,12 +90,31 @@ export class StyleBooksPage {
...
-		await this.page.getByRole('button', {name: buttonText}).click();
+	async selectFrontendTokenCategory(category: string) {
+		await this.page
+			.locator('.style-book-editor__sidebar-content .form-control-select')
+			.click();
 
 		await this.page.getByText(category).click();
 	}
 
+	async updateTokenInput(label: string, value: string, section?: string) {
```
Comment (thiago-buarqque) [inline modules/test/playwright/pages/style-book-web/StyleBooksPage.ts:101]: In this method you call the frontend token just `token`, but in `selectFrontendTokenCategory` you refer to a token as `FrontendToken`

File: modules/test/playwright/tests/style-book-web/stylebookEditor.spec.ts [hunk 2]
```diff
@@ -233,3 +220,242 @@ test('LPD-35561 Preview StyleBook when edit StyleBook', async ({
...
+				addButtonLabel: 'Page',
+				draft: true,
+				name: secondPageName,
+				template: 'Blank',
+			});
+
+			await pageEditorPage.addWidget('Community', 'My Sites');
+		});
+
+		await test.step('Add a paragraph fragment to the third page and publish', async () => {
```
Comment (thiago-buarqque) [inline modules/test/playwright/tests/style-book-web/stylebookEditor.spec.ts:271]: Following the pattern you did for the other two pages

```
Add and publish the third page with a paragraph fragment
```

File: modules/test/playwright/tests/style-book-web/stylebookEditor.spec.ts [hunk 3]
```diff
@@ -233,3 +220,242 @@ test('LPD-35561 Preview StyleBook when edit StyleBook', async ({
...
+			await pageEditorPage.publishPage();
+		});
+
+		await test.step('Add a button fragment to the second page and publish', async () => {
+			await page.getByRole('link', {name: secondPageName}).click();
+
+			await pageEditorPage.addFragment('Basic Components', 'Button');
+
+			await pageEditorPage.publishPage();
+		});
```
Comment (thiago-buarqque) [inline modules/test/playwright/tests/style-book-web/stylebookEditor.spec.ts:300]: Why aren't these two steps done when creating the first and second page?

**Review by thiago-buarqque — APPROVED** [review]
LGTM @pat270. I added three comments, two of them are just suggestions to improve clarity, the other might be an improvement for the test itself (unless there is a reason for it).

Other than that, things are fine

File: modules/test/playwright/tests/style-book-web/stylebookEditor.spec.ts (hunk 3 above)
Comment (pat270) [reply modules/test/playwright/tests/style-book-web/stylebookEditor.spec.ts:300]: @thiago-buarqque I'm not sure. The original poshi test does it this way.

File: modules/test/playwright/pages/style-book-web/StyleBooksPage.ts (hunk 1 above)
Comment (pat270) [reply modules/test/playwright/pages/style-book-web/StyleBooksPage.ts:101]: I named it based on `updateTokenInputColor` just below it. I can change `selectFrontendTokenCategory` to `selectTokenCategory`.

File: modules/test/playwright/tests/style-book-web/stylebookEditor.spec.ts (hunk 3 above)
Comment (pat270) [reply modules/test/playwright/tests/style-book-web/stylebookEditor.spec.ts:300]: Looking at it more, the only thing that makes sense is they wanted to test if drafts were saved.

**Comment by pat270** [comment]
ci:test:sf

**Comment by pat270** [comment]
ci:test:playwright

**Review by thiago-buarqque — APPROVED** [review]
LGTM @ethib137 any thoughts?

**Comment by pat270** [comment]
ci:test:playwright

**Comment by pat270** [comment]
ci:test:playwright

**Comment by pat270** [comment]
ci:forward

**Pull request closed without merging**